session/src/main/scala/akka/cassandra/session/scaladsl/CassandraSessionRegistry.scala

```diff
@@ -81,7 +80,21 @@ final class CassandraSessionRegistry(system: ExtendedActorSystem) extends Extens
       executionContext: ExecutionContext): CassandraSession = {
     val sessionProvider = CqlSessionProvider(system, system.settings.config.getConfig(key.configPath))
     val log = Logging(system, classOf[CassandraSession])
-    new CassandraSession(system, sessionProvider, executionContext, log, metricsCategory = key.configPath, init)
+    new CassandraSession(
+      system,
+      sessionProvider,
+      executionContext,
+      log,
+      metricsCategory = key.configPath,
+      init,
+      onClose = () => sessions.remove(key))
   }
 
+  private def close(): Future[Done] = {
+    import system.dispatcher
+    val closing = sessions.values().asScala.map(_.close())
+    Future.sequence(closing).map(_ => Done)
+  }
+
+  system.whenTerminated.foreach(_ => close())(system.dispatcher)
```

Comment:
have you verified that it's possible to use the `system.dispatcher` after the ActorSystem has been terminated?

incidentally there is a question about that in https://discuss.lightbend.com/t/registerontermination-for-akka-typed-actor/5783/4

Reply:
Good point. That would mean this shutdown should go into `CoordinatedShutdown` instead, I suggest.

Reply:
That is not very good, since actors may want to use the session late in the shutdown sequence.
I see two options
1) run it with Scala's global
2) run it with `sameThreadExecutionContext`, but then it's important to wrap the `close` with `Try` so that any exception is swallowed